core/trino-main/src/main/java/io/trino/operator/GroupByIdBlock.java

```diff
@@ -156,27 +227,57 @@ public Block copyWithAppendedNull()
     @Override
     public String toString()
     {
-        return toStringHelper(this)
-                .add("groupCount", groupCount)
-                .add("positionCount", getPositionCount())
-                .toString();
+        return isRle() ?
+                "RLE(" + getGroupId(0) + ")" :
+                toStringHelper(this)
+                        .add("groupCount", groupCount)
+                        .add("positionCount", getPositionCount())
+                        .toString();
     }
 
     @Override
     public boolean isLoaded()
     {
-        return block.isLoaded();
+        return true;
     }
 
     @Override
     public Block getLoadedBlock()
     {
-        return block.getLoadedBlock();
+        return this;
     }
 
     @Override
     public final List<Block> getChildren()
     {
-        return singletonList(block);
+        return singletonList(this);
+    }
+
+    @Override
+    public Block getPositions(int[] positions, int offset, int length)
+    {
+        return copyPositions(positions, offset, length);
+    }
+
+    private void checkReadablePosition(int position)
+    {
+        if (position < 0 || position >= getPositionCount()) {
+            throw new IllegalArgumentException("position is not valid");
```

Comment:
Improve exception message as in `io.trino.spi.block.BlockUtil#checkValidPosition`